### PR TITLE
Remove step states from snapshots

### PR DESCRIPTION
**Pull request by berland**
The step state is moved upwards to the realization.

**Issue**
Partially solves #6218 

**Approach**
Cut code,  and merge datastructure.

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/ensemble_evaluator/identifiers.py [hunk 1]
```diff
@@ -17,28 +17,28 @@
...
 STEPS = "steps"
 
-EVTYPE_FM_STEP_FAILURE = "com.equinor.ert.forward_model_step.failure"
-EVTYPE_FM_STEP_PENDING = "com.equinor.ert.forward_model_step.pending"
-EVTYPE_FM_STEP_RUNNING = "com.equinor.ert.forward_model_step.running"
-EVTYPE_FM_STEP_SUCCESS = "com.equinor.ert.forward_model_step.success"
-EVTYPE_FM_STEP_UNKNOWN = "com.equinor.ert.forward_model_step.unknown"
-EVTYPE_FM_STEP_WAITING = "com.equinor.ert.forward_model_step.waiting"
-EVTYPE_FM_STEP_TIMEOUT = "com.equinor.ert.forward_model_step.timeout"
+EVTYPE_REALIZATION_FAILURE = "com.equinor.ert.realization.failure"
```
Comment (xjules) [inline src/ert/ensemble_evaluator/identifiers.py:20]: These are the new realization statuses.

File: src/ert/ensemble_evaluator/snapshot.py [hunk 2]
```diff
@@ -118,18 +103,18 @@ def update_metadata(self, metadata: Dict[str, Any]) -> None:
...
-        step_update = _filter_nones(
-            {
-                "status": step.status,
-                "start_time": step.start_time,
-                "end_time": step.end_time,
-            }
+    def update_realization(
+        self,
+        real_id: str,
+        status: Optional[str] = None,
```
Comment (xjules) [inline src/ert/ensemble_evaluator/snapshot.py:109]: Is this fine that it can be `None`?

File: src/ert/job_queue/queue.py [hunk 3]
```diff
@@ -50,30 +50,30 @@
...
 
-
-_FM_STEP_FAILURE = "com.equinor.ert.forward_model_step.failure"
-_FM_STEP_PENDING = "com.equinor.ert.forward_model_step.pending"
-_FM_STEP_RUNNING = "com.equinor.ert.forward_model_step.running"
-_FM_STEP_SUCCESS = "com.equinor.ert.forward_model_step.success"
-_FM_STEP_UNKNOWN = "com.equinor.ert.forward_model_step.unknown"
-_FM_STEP_WAITING = "com.equinor.ert.forward_model_step.waiting"
-
+# Duplicated from ert.ensemble_evaluator.identifiers to avoid circular imports
```
Comment (xjules) [inline src/ert/job_queue/queue.py:53]: What about putting those `identifiers.py` into some standalone module?

File: tests/unit_tests/ensemble_evaluator/ensemble_evaluator_utils.py [hunk 4]
```diff
@@ -103,13 +103,6 @@ def _evaluate(self, url):
             event_id = event_id + 1
             for real in range(0, self.test_reals):
                 job_failed = False
-                send_dispatch_event(
```
Comment (xjules) [inline tests/unit_tests/ensemble_evaluator/ensemble_evaluator_utils.py:106]: Should we send a corresponding `REALIZATION_UNKNOW` status event?

**Review by xjules — CHANGES_REQUESTED** [review]
Looks very good! Just had a small comment here and there. Mainly though, if we can move the `identifiers.py`?

File: src/ert/ensemble_evaluator/identifiers.py (hunk 1 above)
Comment (berland) [reply src/ert/ensemble_evaluator/identifiers.py:20]: I think I will call these *renamed* events.  There were realization *states* before, but they were implicitly updated based on the step states by the function `_propagate_step_update_to_realization`. Now we update them directly using their own events.

File: src/ert/job_queue/queue.py (hunk 3 above)
Comment (berland) [reply src/ert/job_queue/queue.py:53]: That could probably solve the circular uglyness yes, but I think that should be its own PR :)

Comment (berland) [reply src/ert/job_queue/queue.py:53]: https://github.com/equinor/ert/issues/6457

File: src/ert/ensemble_evaluator/snapshot.py (hunk 2 above)
Comment (berland) [reply src/ert/ensemble_evaluator/snapshot.py:109]: Hmm, that would be if you have an update for start- or end-time without having an update for the state. Not entirely sure if this is a scenario that actually occurs.

Comment (berland) [reply src/ert/ensemble_evaluator/snapshot.py:109]: You are right, it cannot happen that `status is None`. Will fix.

File: tests/unit_tests/ensemble_evaluator/ensemble_evaluator_utils.py (hunk 4 above)
Comment (berland) [reply tests/unit_tests/ensemble_evaluator/ensemble_evaluator_utils.py:106]: Added that. But don't know what it is good for.

File: src/ert/ensemble_evaluator/snapshot.py (hunk 2 above)
Comment (berland) [reply src/ert/ensemble_evaluator/snapshot.py]: Now changed. This implies that a `dict.get()` in `from_cloudevent` had to be changed to a direct accessor which can yield a `KeyError`. A buggy job_runner can then crash the evaluator by sending unknown events.

File: src/ert/job_queue/queue.py (hunk 3 above)
Comment (berland) [reply src/ert/job_queue/queue.py]: (the duplication was there prior to this PR)

File: src/ert/ensemble_evaluator/snapshot.py (hunk 2 above)
Comment (berland) [reply src/ert/ensemble_evaluator/snapshot.py]: test_dark_storage triggered this KeyError. Fixed by not mapping event type to state before we know what kind of event we are talking about.

**Review by xjules — APPROVED** [review]
Looks good now! Could you squash / split commits?

**Pull request merged**